src/main/kotlin/com/demonwav/mcdev/i18n/i18n-utils.kt

```diff
@@ -0,0 +1,57 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2017 minecraft-dev
+ *
+ * MIT License
+ */
+
+package com.demonwav.mcdev.i18n
+
+import com.demonwav.mcdev.i18n.lang.I18nFile
+import com.demonwav.mcdev.i18n.lang.I18nFileType
+import com.demonwav.mcdev.i18n.lang.gen.psi.I18nProperty
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiManager
+import com.intellij.psi.search.FileTypeIndex
+import com.intellij.psi.search.GlobalSearchScope
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.util.indexing.FileBasedIndex
+
+enum class Scope {
+    GLOBAL, PROJECT
+}
+
+private fun Project.files(scope: Scope) =
+    FileBasedIndex.getInstance().getContainingFiles(FileTypeIndex.NAME, I18nFileType, if (scope == Scope.GLOBAL) GlobalSearchScope.allScope(this) else GlobalSearchScope.projectScope(this))
+        .map { PsiManager.getInstance(this).findFile(it) as I18nFile? }
+        .filter { it != null }
+        .map { it as I18nFile }
+
+private fun Project.findPropertiesImpl(scope: Scope, fileFilter: (I18nFile) -> Boolean = { true }, propertyFilter: (I18nProperty) -> Boolean = { true }) =
+    files(scope)
+        .filter(fileFilter)
+        .flatMap { (PsiTreeUtil.getChildrenOfType(it, I18nProperty::class.java) ?: emptyArray()).asIterable() }
```

Comment:
Something like `getChildrenOfType()?.asIterable() ?: emptyList()` would be more efficient because you avoid creating the empty array and the iterable if there are no children.